lib/irb/debug.rb

```diff
@@ -57,6 +57,19 @@ def DEBUGGER__.capture_frames(*args)
           DEBUGGER__::ThreadClient.prepend(SkipPathHelperForIRB)
         end
 
+        Reline.output_modifier_proc = proc do |output, _|
+          next output if output.strip.empty?
+
+          cmd = output.split(/\s/, 2).first
+
+          if DEBUGGER__.commands.keys.grep(/\A#{Regexp.escape(cmd)}/).any?
```

Comment:
`w = 1` is not a command but debug hint is show. 
```
From: a.rb @ line 1 :

 => 1: binding.irb

irb(main):001> debug
irb:rdbg(main):002> w = 1 # debug command
```
How about `DEBUGGER__.commands.key?(cmd)`